### DISCUSSION TITLE
Remove multiple PAS users

### DISCUSSION THREAD

**Discussion by vamshieee** [pspete/psPAS#425] [Ideas]
Hi All,

I would like to user Remove-PASUser command to delete multiple users. Could you please validate this.

New-PASSession -Credential $cred -BaseURI https://pvwa/ -type RADIUS

$users= Get-content 'C:\Users\localadmin\Desktop\DeleteUsers.csv'

 

foreach($user in $users)

{

  echo "removing $user"

 

  Remove-PASUser -UserName $user

}

**Comment by pspete** [comment]
Not enough detail provided. 
What validation are you needing?
What version do you use? Module version/ cyberark version? 
What Generation of API do you target?
What's in the csv? What does '$user' evaluate to in the loop?
Have you tried the '-whatif' parameter?

**Comment by vamshieee** [comment]
Hi @pspete I have a list of inactive users who are not using CyberArk for 45 days. I want to remove all of them at a glance instead of single user at a time.
Will this work fine for removing multiple users. Because I need to be sure this works fine as I don't have test env for testing
I am using CyberArk version 12.2
CSV consist of users list and $user evaluate to single EPV user that to be removed from Vault.
No, I haven't tried '-whatif' parameter.

**Reply by pspete** [reply]
Advice is to ensure your code works in a test, non-prod, or other environment before running against a production solution. It is your responsibility to make sure the code does what you intend.

Assuming the latest version of the psPAS module is being used:
- For 12.2, You can use the Gen 2 API.
  - The syntax is in the documentation here: https://pspas.pspete.dev/commands/Remove-PASUser#gen2-default
  - Integer `id` value for the user must be provided instead of `username`.

If you are using the Gen 1 API (Syntax here: https://pspas.pspete.dev/commands/Remove-PASUser#gen1)
- `UserName` can be specified
- The [`-WhatIf` paramter](https://pspas.pspete.dev/commands/Remove-PASUser#-whatif) can be used to show what would happen if the command is run, without actually actioning the remove user command. i.e.:
```powershell 
Remove-PASUser -UserName SomeUser -WhatIf
WhatIf: Performing the operation "Delete User" on target "SomeUser"
```
